runtime/lua/vim/lsp/completion.lua

```diff
@@ -671,6 +671,107 @@ local function get_augroup(bufnr)
   return string.format('nvim.lsp.completion_%d', bufnr)
 end
 
+---@param winid integer
+---@param bufnr integer
+---@param ft? string
+local function update_popup_window(winid, bufnr, ft)
+  if winid and api.nvim_win_is_valid(winid) and bufnr and api.nvim_buf_is_valid(bufnr) then
+    vim.wo[winid].conceallevel = 2
+    if ft and vim.bo[bufnr].filetype ~= ft then
+      vim.bo[bufnr].filetype = ft
+    end
+    local all = api.nvim_win_text_height(winid, {}).all
+    api.nvim_win_set_height(winid, all)
+  end
+end
+
+---@return function
+local function cmpitem_resolve()
+  local infoContext = {
+    timer = nil, -- [[uv_time_t]]
+    cancel_funcs = {}, --- @type function[]
+    bufnr = nil,
+    word = nil,
+  }
+
+  local function cancel_pending()
+    for _, cancel in ipairs(infoContext.cancel_funcs) do
+      cancel()
+    end
+    infoContext.cancel_funcs = {}
+  end
+
+  local function cleanup()
+    if infoContext.timer and not infoContext.timer:is_closing() then
+      infoContext.timer:stop()
+      infoContext.timer:close()
+      infoContext.timer = nil
+    end
+    cancel_pending()
+  end
+
+  local function is_valid()
+    local data = vim.fn.complete_info({ 'selected', 'completed' })
+    return api.nvim_buf_is_valid(infoContext.bufnr)
+      and api.nvim_get_current_buf() == infoContext.bufnr
+      and vim.startswith(api.nvim_get_mode().mode, 'i')
+      and tonumber(vim.fn.pumvisible()) == 1
+      and (vim.tbl_get(data, 'completed', 'word') or '') == infoContext.word
+  end
+
+  return function(bufnr, param, select_word)
+    cleanup()
+    infoContext.bufnr = bufnr
+    infoContext.word = select_word
+
+    infoContext.timer = assert(vim.uv.new_timer())
+    infoContext.timer:start(
+      100,
+      0,
+      vim.schedule_wrap(function()
+        if not is_valid() then
+          cleanup()
+          return
+        end
+
+        cancel_pending()
+
+        local _, cancel_func = vim.lsp.buf_request(
```

Comment:
can we send this request only to the specific `user_data.nvim.lsp.client_id` that provided the completion item? Currently it sends request to all of them, even the clients that were not involved with the completion item. I tried out this branch and it causes a bit of flicker/chaos when there is more than one LSP server attached.

Reply:
this field only in Lsp relate autocmd